builtin/providers/aws/auth_helpers.go

```diff
@@ -137,7 +142,49 @@ func GetCredentials(c *Config) *awsCredentials.Credentials {
 		}
 	}
 
-	return awsCredentials.NewChainCredentials(providers)
+	if c.AssumeRoleARN != "" {
+		log.Printf("[INFO] Attempting to AssumeRole %s (SessionName: %q, ExternalId: %q)",
+			c.AssumeRoleARN, c.AssumeRoleSessionName, c.AssumeRoleExternalID)
+
+		creds := awsCredentials.NewChainCredentials(providers)
+		cp, err := creds.Get()
+		if err != nil {
+			if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == "NoCredentialProviders" {
+				errs = append(errs, errors.New(`No valid credential sources found for AWS Provider.
```

Comment:
I think this is the error message we spoke about - isn't this the same generic error message we throw for non-assume role? Shouldn't we mention something here about assume role not working due to credentials?